src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -348,7 +358,14 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
 
         aMessage.Read(offset, sizeof(Tlv), &data.tlv);
         aMessage.Read(offset + sizeof(Tlv), data.tlv.GetLength(), data.value);
-        mLocal.Set(data.tlv);
+
+        if (data.tlv.GetType() != Tlv::kChannel && data.tlv.GetType() != Tlv::kPanId &&
+            data.tlv.GetType() != Tlv::kNetworkMasterKey &&
+            data.tlv.GetType() != Tlv::kMeshLocalPrefix)
+        {
+            mLocal.Set(data.tlv);
```

Comment:
Yes, a change to the Active Operational Dataset can only be done by receiving a new Active Operational Dataset via a MGMT_ACTIVE_SET request message.  As I mentioned before, the change needs to be all or nothing.  If you break up the operational dataset, then you have the same Active Dataset Timestamp in both the Active and Pending Operational Datasets, however the Active Dataset Timestamp represents different datasets.  When the Pending Delay Timer expires, the Pending Dataset cannot be applied to the Active Dataset because the Active Timestamp does not advance.  As a result, if you are delaying a change to the Active Operational Dataset, you need to delay the entire dataset (not just a portion of it).